test/integ/test_key_pair_authentication.py

```diff
@@ -5,13 +5,84 @@
 #
 
 import uuid
+from datetime import datetime, timedelta
+from os import path
 
+import jwt
 import pytest
 from cryptography.hazmat.backends import default_backend
 from cryptography.hazmat.primitives import serialization
+from cryptography.hazmat.primitives._serialization import (
+    Encoding,
+    NoEncryption,
+    PrivateFormat,
+)
 from cryptography.hazmat.primitives.asymmetric import dsa, rsa
+from cryptography.hazmat.primitives.serialization import (
+    load_der_private_key,
+    load_pem_private_key,
+)
 
 import snowflake.connector
+from snowflake.connector.auth import get_token_from_private_key
+from snowflake.connector.auth_keypair import AuthByKeyPair
+
+
+@pytest.mark.skipolddriver
+def test_get_token_from_private_key():
```

Comment:
This test case generates a jwt token using the new utility function.
It then decodes the token and asserts that the fields (iss, sub, exp) contain the correct values.

I do not see a way to make a python connection with the token without changing a lot of the python connector logic, which seems too risky just for a test case. However, manual testing of the tokens using the sqlapi interface shows that they work. The test cases for making a connection with the token will be in the t_sqlapi regression test library.